### PR TITLE
Proposal update reviewer checklist

### PR DESCRIPTION
**Pull request by anneliesdej**
**Current status**: Complete and ready for full review
The checklist for review is updated to give an ordering that better facilitates big to small changes. The checklist for developer is updated to include asking for a partial review.

### CODE REVIEW

File: docs/source/developer/checklist_for_developer.md [hunk 1]
```diff
@@ -49,6 +49,10 @@ Use this checklist as a practical working guide during development. It is not me
 - [ ] The pull request merges into develop.
 - [ ] The pull request title is descriptive and includes relevant issue references
 - [ ] The pull request description clearly explains what changed, why it changed, the scope of the change and any known limitations or areas requiring attention
+- [ ] If no scope is specified, the reviewer assumes a full review is requested. If requesting a partial review, the scope is explicitly defined in the pull request description. It is clearly stated:
+  - [ ] which parts should be reviewed
+  - [ ] which parts are not ready for review
+  - [ ] what kind of feedback is expected (e.g. design, approach, API)
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_developer.md]:
```suggestion
  - [ ] what kind of feedback is expected (e.g. design, approach, adherence to conventions, usability)
```

File: docs/source/developer/checklist_for_review.md [hunk 2]
```diff
@@ -3,121 +3,144 @@ This checklist is intended to guide reviewers in performing consistent, thorough
...
+This checks whether the implementation is fully and meaningfully validated across all functional paths and data flows.
+- [ ] All implemented functionality is covered by tests
 - [ ] Both success and failure cases are tested
-- [ ] Edge cases from the specification are explicitly tested
-- [ ] Regression tests are added where relevant (previous bugs stay fixed)
-### Standard test dimensions
+- [ ] Edge cases from the functional description are explicitly tested
+- [ ] Correct stubs, mocks and spies are used where applicable in the tests
+- [ ] Real-world usage scenarios are represented in tests
+- [ ] Regression tests are added where relevant
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_review.md]: Regression testing is by definition on code that already exsists. Did you mean integration testing?
It might be good to mention that integration tests are usually needed when mocks/stubs are used in unit testing.

File: docs/source/developer/checklist_for_review.md [hunk 3]
```diff
@@ -3,121 +3,144 @@ This checklist is intended to guide reviewers in performing consistent, thorough
...
+- [ ] Critical paths are explicitly tested, not only indirectly covered
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_review.md]: What do you mean with "critical path" ?

File: docs/source/developer/checklist_for_review.md [hunk 4]
```diff
@@ -3,121 +3,144 @@ This checklist is intended to guide reviewers in performing consistent, thorough
...
-- [ ] Single-element inputs behave correctly
-- [ ] Extreme numeric values are handled safely
 - [ ] Division by zero is handled or prevented
-- [ ] Invalid inputs fail in a predictable and informative way
+- [ ] Optional parameters have sensible and safe defaults
 - [ ] Behaviour is stable across repeated executions
 
-## 6. Scikit-learn estimator compliance (if applicable)
+## 7. Scikit-learn estimator compliance (if applicable)
+This checks whether the implementation conforms to scikit-learn's API and behavioural contract, if applicable.
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_review.md]: perhaps add a link to the sklearn developer docs?

File: docs/source/developer/checklist_for_review.md [hunk 5]
```diff
@@ -3,121 +3,144 @@ This checklist is intended to guide reviewers in performing consistent, thorough
...
 - [ ] Estimator passes `check_estimator` or `parametrize_with_checks`
 - [ ] `fit`, `transform`, etc. behave correctly and consistenly
 - [ ] `get_params` and `set_params` work correctly
 - [ ] Estimator is stateless where required (e.g. `transform`)
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_review.md]: When is an estimator required to be stateless? Most estimators have at least 2 states: not fitted, fitted.

File: docs/source/developer/checklist_for_developer.md (hunk 1 above)
Comment (KoomenErina) [reply docs/source/developer/checklist_for_developer.md]: perhaps refer/link to the API section of the review checklist?

**Review by KoomenErina — CHANGES_REQUESTED** [review]

File: docs/source/developer/checklist_for_review.md (hunk 2 above)
Comment (anneliesdej) [reply docs/source/developer/checklist_for_review.md]: Removed regression testing. Integration tests is already mentioned

File: docs/source/developer/checklist_for_review.md (hunk 5 above)
Comment (anneliesdej) [reply docs/source/developer/checklist_for_review.md]: stateless is not the correct wording here, I changed it

File: docs/source/developer/checklist_for_review.md [hunk 6]
```diff
@@ -3,121 +3,142 @@ This checklist is intended to guide reviewers in performing consistent, thorough
...
 - [ ] Complexity is acceptable for intended use cases
 
-## 8. Documentation and usability
-- [ ] Docstrings are accurate and reflect actual behaviour
-- [ ] Examples are included for non-trivial functionality
-- [ ] Examples are correct and executable
-- [ ] Terminology is consistent with project standards
-
 ## 9. Form and style
+This checks whether the code is readable, consistent and aligned with project-wide conventoins.
```
Comment (KoomenErina) [inline docs/source/developer/checklist_for_review.md]:
```suggestion
This checks whether the code is readable, consistent and aligned with project-wide conventions.
```

File: docs/source/developer/checklist_for_developer.md
Comment (KoomenErina) [inline docs/source/developer/checklist_for_developer.md]: this does not build on readthedocs

**Review by KoomenErina — CHANGES_REQUESTED** [review]

**Review by KoomenErina — APPROVED** [review]
